common/static/sass/edx-pattern-library-shims/base/_variables.scss

```diff
@@ -178,6 +178,22 @@ $btn-brand-active-background:             $uxpl-blue-base !default;
 $btn-brand-disabled-background:           #f2f3f3 !default;
 $btn-brand-disabled-color:                #676666 !default;
 
+// Upgrade button
+$btn-upgrade-border-color:                $uxpl-green-base !default;
```

Comment:
I agree that this is not an official button pattern yet, so it shouldn't be in this file. In an ideal world it would be a pattern, because we use green buttons in a fairly consistent way, but we haven't reached that level of agreement yet.

Regarding the intermediary variables: we should not use colors directly in Sass files because it makes it very difficult for someone to theme. In Sass we shouldn't be saying that a button is green, we should be saying that it is an upgrade button. Then a theme can just override the color for upgrade buttons in one place and see it applied everywhere. If instead we have rules everywhere that specify that particular buttons are green, then a theme developer has to find every single one and override it. Worse still, we can in the future change the rules or introduce new ones, and the theme will break.

We need to have a Sass best practices wiki describing some of these issues. I've been putting it off, but I think I need to start with something in the wiki so that we're all on the same page.